utils/src/main/kotlin/Utils.kt

```diff
@@ -567,3 +571,46 @@ fun Throwable.showStackTrace() {
     // https://discuss.kotlinlang.org/t/if-operator-in-function-expression/7227.
     if (printStackTrace) printStackTrace()
 }
+
+/**
+ * Return a Closable which temporarily moves the given directories to a temporary directory on initialization. On close
+ * it moves them back to their original location overwriting any conflicting existing directory.
+ */
+fun stashDirectories(vararg directories: File): Closeable = StashDirectory(setOf(*directories))
+
+private class StashDirectory(directories: Set<File>) : Closeable {
+    private val stashedDirectories = mutableMapOf<File, File>()
+
+    init {
+        directories.forEach {
```

Comment:
Esp. when doing the inlining below, I'd inline this, too and write

        directories.filter {
            it.isDirectory
        }.associateTo(stashedDirectories) { originalDir ->
            val tempDir = createTempDir(originalDir.name, ".tmp", originalDir.parentFile)
            log.info { "Temporarily moving directory from '${originalDir.absolutePath}' to '${tempDir.absolutePath}'." }

            Files.move(originalDir.toPath(), tempDir.toPath(), StandardCopyOption.ATOMIC_MOVE)
            Pair(originalDir, tempDir)
        }

Reply:
mhh, that seems much nicer